crates/nu-protocol/src/ast/unit.rs

```diff
@@ -35,6 +35,20 @@ pub enum Unit {
     Week,
 }
 
+/// TODO: something like `Filesize::from_unit` in the future?
+fn duration_mul_and_check(size: i64, factor: i64, span: Span) -> Result<Value, ShellError> {
+    match size.checked_mul(factor) {
+        Some(val) => Ok(Value::duration(val, span)),
+        None => Err(ShellError::GenericError {
+            error: "duration too large".into(),
```

Comment:
For @cptpiepmatz 's war on `GenericError` maybe `InvalidValue` or one of the overflow variants could fit this application.